registry/storage/driver/s3-aws/s3.go

```diff
@@ -238,31 +250,59 @@ func FromParameters(parameters map[string]interface{}) (*Driver, error) {
 // New constructs a new Driver with the given AWS credentials, region, encryption flag, and
 // bucketName
 func New(params DriverParameters) (*Driver, error) {
-	awsConfig := aws.NewConfig()
-	creds := credentials.NewChainCredentials([]credentials.Provider{
-		&credentials.StaticProvider{
-			Value: credentials.Value{
-				AccessKeyID:     params.AccessKey,
-				SecretAccessKey: params.SecretKey,
+	var s3obj *s3.S3
+	if params.Backend == "minio" {
+		newSession := session.New()
+		s3Config := &aws.Config{
+			Credentials:      credentials.NewStaticCredentials(params.AccessKey, params.SecretKey, ""),
+			Endpoint:         aws.String(params.EndpointURL),
+			Region:           aws.String(params.Region),
+			DisableSSL:       aws.Bool(true),
+			S3ForcePathStyle: aws.Bool(true),
+		}
+		// Create an S3 service object in the default region.
+		s3obj = s3.New(newSession, s3Config)
+	} else {
+		awsConfig := aws.NewConfig()
+		creds := credentials.NewChainCredentials([]credentials.Provider{
+			&credentials.StaticProvider{
+				Value: credentials.Value{
+					AccessKeyID:     params.AccessKey,
+					SecretAccessKey: params.SecretKey,
+				},
 			},
-		},
-		&credentials.EnvProvider{},
-		&credentials.SharedCredentialsProvider{},
-		&ec2rolecreds.EC2RoleProvider{Client: ec2metadata.New(session.New())},
-	})
+			&credentials.EnvProvider{},
+			&credentials.SharedCredentialsProvider{},
+			&ec2rolecreds.EC2RoleProvider{Client: ec2metadata.New(session.New())},
+		})
 
-	awsConfig.WithCredentials(creds)
-	awsConfig.WithRegion(params.Region)
-	awsConfig.WithDisableSSL(!params.Secure)
-	// awsConfig.WithMaxRetries(10)
+		awsConfig.WithCredentials(creds)
+		awsConfig.WithRegion(params.Region)
+		awsConfig.WithDisableSSL(!params.Secure)
 
-	if params.UserAgent != "" {
-		awsConfig.WithHTTPClient(&http.Client{
-			Transport: transport.NewTransport(http.DefaultTransport, transport.NewHeaderRequestModifier(http.Header{http.CanonicalHeaderKey("User-Agent"): []string{params.UserAgent}})),
-		})
+		// awsConfig.WithMaxRetries(10)
+
+		if params.UserAgent != "" {
+			awsConfig.WithHTTPClient(&http.Client{
+				Transport: transport.NewTransport(http.DefaultTransport, transport.NewHeaderRequestModifier(http.Header{http.CanonicalHeaderKey("User-Agent"): []string{params.UserAgent}})),
+			})
+		}
+
+		s3obj = s3.New(session.New(awsConfig))
 	}
 
-	s3obj := s3.New(session.New(awsConfig))
+	_, err := s3obj.CreateBucket(&s3.CreateBucketInput{
+		Bucket: &params.Bucket,
+	})
+	if err != nil {
+		if s3err, ok := err.(awserr.Error); ok && s3err.Code() != "BucketAlreadyOwnedByYou" {
```

Comment:
I am ignoring BucketAlreadyOwnedByYou error and throwing error otherwise. We want the user to know if BucketAlreadyExists and he doesn't own it,hence i am throwing the error.